presto-base-jdbc/src/main/java/io/prestosql/plugin/jdbc/SliceReadFunction.java

```diff
@@ -30,4 +30,15 @@
 
     Slice readSlice(ResultSet resultSet, int columnIndex)
             throws SQLException;
+
+    @Override
+    default boolean isNull(ResultSet resultSet, int columnIndex)
+            throws SQLException
+    {
+        // JDBC is kind of dumb: we need to read the field and then ask
+        // if it was null, which means we are wasting effort here.
+        // We could save the result of the field access if it matters.
+        resultSet.getString(columnIndex);
```

Comment:
I do not see a reason why resultSet.getString() should be better than resultSet.getObject().
The code is redundant and can easily be accidentally removed with a refactor.
Can we have a test for this?

Also, if this is cockrach specific behavior, the correct approach is to provide `SliceReadFunction` and implement `isNull` for that case.